### ISSUE TITLE
Option Shift text selection on Mac OS

### ISSUE THREAD

**Issue by jish** [zed-industries/zed#10242]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Selecting text using the option key, shift, and the arrow keys is a common part of my text editing workflow. I feel that the selection behavior is not quite right when using the `"base_keymap": "TextMate"` configuration.

Here is an example of the behavior in Zed when pressing `option` + `shift` + `left-arrow`:

https://github.com/zed-industries/zed/assets/1817/6aa80bfb-d7b5-43b4-8545-b67119835df6

Here is the exact same source code file, and keystrokes in TextMate:

https://github.com/zed-industries/zed/assets/1817/87c5032d-8bb3-4965-a452-cf822b7e3d60

Now, if I change base keymap setting to use Sublime instead `"base_keymap": "SublimeText"` then I get slightly different behavior:

https://github.com/zed-industries/zed/assets/1817/8481bfd5-9ec9-4528-95f0-fb0250c26029

### Environment

Zed: v0.129.2 (Zed)
OS: macOS 14.1.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

I believe selecting text with option shift and arrow keys should have logical stops to progressively select more and more text, instead of selecting the entire line.

https://github.com/zed-industries/zed/assets/1817/b34c19e5-5bee-46c6-bd34-dbd1dac3a6da

**Comment by jish** [comment]
I can solve this issue by copying the `"alt-shift-left"` and `"alt-shift-right"` configurations from Default Key Bindings to my `~/.config/zed/keymap.json`:

```json
[
  {
    "context": "Editor",
    "bindings": {
      "alt-shift-left": "editor::SelectToPreviousWordStart",
      "alt-shift-right": "editor::SelectToNextWordEnd"
    }
  }
]
```

Maybe this could be solved by removing the `"alt-shift-left"` and `"alt-shift-right"` overrides from `assets/keymaps/textmate.json`?

https://github.com/zed-industries/zed/blob/c7961b9054d73155d1a6ab9cd26c232468e4098f/assets/keymaps/textmate.json#L41-L52

**Comment by JosephTLyons** [comment]
CC'ing in @caius, as I think they did the work adding in the TextMate keymap originally (when they were in the `keymaps` repository):

- https://github.com/zed-industries/keymaps/pull/16

If TextMate ships with the behavior of jumping by words, rather than to the beginning of the line, we should change it to that.  Hey @caius, curious what you think here. Caius did the [work adding in the TextMate keymap](https://github.com/zed-industries/keymaps/pull/16) originally (when they keymaps were in the `keymaps` repository.

**Comment by JosephTLyons** [comment]
Just checked - I downloaded TextMate and its default behavior is as described by @jish.

**Comment by JosephTLyons** [comment]
Would want want to simply remove those? The default bindings use the same keys, but they do not explicitly set `"stop_at_soft_wraps": true`, as these do in the TextMate keymap.

At any rate, I've marked this as a `good first issue` - feel free to open a PR here, @jish. :)

**Comment by JosephTLyons** [comment]
This should be fixed now in [v0.135.0-pre](https://zed.dev/releases/preview/0.135.0).
